cloud_experts_tutorials/cloud-experts-using-cloudfront-and-waf.adoc

```diff
@@ -0,0 +1,359 @@
+
+. Create a AWS WAF rules file to apply to our web ACL:
++
+[source,terminal]
+----
+$ cat << EOF > ${SCRATCH}/waf-rules.json 
+[
+    {
+      "Name": "AWS-AWSManagedRulesCommonRuleSet",
+      "Priority": 0,
+      "Statement": {
+        "ManagedRuleGroupStatement": {
+          "VendorName": "AWS",
+          "Name": "AWSManagedRulesCommonRuleSet"
+        }
+      },
+      "OverrideAction": {
+        "None": {}
+      },
+      "VisibilityConfig": {
+        "SampledRequestsEnabled": true,
+        "CloudWatchMetricsEnabled": true,
+        "MetricName": "AWS-AWSManagedRulesCommonRuleSet"
+      }
+    },
+    {
+      "Name": "AWS-AWSManagedRulesSQLiRuleSet",
+      "Priority": 1,
+      "Statement": {
+        "ManagedRuleGroupStatement": {
+          "VendorName": "AWS",
+          "Name": "AWSManagedRulesSQLiRuleSet"
+        }
+      },
+      "OverrideAction": {
+        "None": {}
+      },
+      "VisibilityConfig": {
+        "SampledRequestsEnabled": true,
+        "CloudWatchMetricsEnabled": true,
+        "MetricName": "AWS-AWSManagedRulesSQLiRuleSet"
+      }
+    }
+]
+EOF
+----
++
+This will enable the Core (Common) and SQL AWS Managed Rule Sets.
++
+. Create an AWS WAF Web ACL using the rules we specified above:
++
+[source,terminal]
+----
+$ WAF_WACL=$(aws wafv2 create-web-acl \
+  --name cloudfront-waf \
+  --region ${REGION} \
+  --default-action Allow={} \
+  --scope CLOUDFRONT \
+  --visibility-config SampledRequestsEnabled=true,CloudWatchMetricsEnabled=true,MetricName=${CLUSTER_NAME}-waf-metrics \
+  --rules file://${SCRATCH}/waf-rules.json \
+  --query 'Summary.Name' \
+  --output text)
+----
+
+[id="configure_amazon_cloudfront_{context}"]
+== Configure Amazon CloudFront
+
+. Retrieve the newly created custom domain ingress controller's NLB hostname:
++
+[source,terminal]
+----
+$ NLB=$(oc -n openshift-ingress get service router-cloudfront-waf \
+  -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')
+$ echo "Origin domain: ${NLB}"
+----
+
+. Import your certificate into Amazon Certificate Manager, where `cert.pem` is your wildcard certificate, `fullchain.pem` is your wildcard certificate's chain and `privkey.pem` is your wildcard certificate’s private key.
++
+[NOTE]
+====
+Regardless of what region your cluster is deployed, you must import this certificate to `us-east-1` as Amazon CloudFront is a global AWS service.
+====
++
+.Example
+[source,terminal]
+----
+$ aws acm import-certificate --certificate file://cert.pem \
+  --certificate-chain file://fullchain.pem \
+  --private-key file://privkey.pem \
+  --region us-east-1
+----
+
+. Log into the link:https://us-east-1.console.aws.amazon.com/cloudfront/v3/home#/distributions/create[AWS console] to create a CloudFront distribution.
++
+. Configure the CloudFront distribution by using the following information:
++
+[NOTE]
+====
+If an option is not specified in the table below, leave them the default (which may be blank).
```

Comment:
Suggestion:
If an option is not specified in the table below, leave them the default, which may be blank.